src/pint/exceptions.py

```diff
@@ -0,0 +1,162 @@
+__all__ = [
+    "TimingModelError",
+    "MissingParameter",
+    "MissingTOAs",
+    "MissingBinaryError",
+    "UnknownBinaryModel",
+]
+
+
+# originally from fitter.py
+class DegeneracyWarning(UserWarning):
+    pass
+
+
+class ConvergenceFailure(ValueError):
+    pass
+
+
+class MaxiterReached(ConvergenceFailure):
+    pass
+
+
+class StepProblem(ConvergenceFailure):
+    pass
```

Comment:
Can you please add docstrings explaining what each of these exceptions mean?

Reply:
I don't think I necessarily can, since I didn't write them.  I could probably infer some from usage but that would take a while.